### PR TITLE
docs: Clarify using Config, especially params that can be zero

### LINKED ISSUE #1250: Clarify which configuration options are mandatory
Using `quiche::Config::default()` will almost never work. At least the following options have to be set:
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_initial_max_streams_bidi
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_initial_max_streams_uni
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_initial_max_data
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_initial_max_stream_data_bidi_local
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_initial_max_stream_data_bidi_remote
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_initial_max_stream_data_uni

In most cases, the ALPN and certificates have to be set as well:
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.set_application_protos
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.load_verify_locations_from_file (for clients)
- https://docs.rs/quiche/latest/quiche/struct.Config.html#method.load_cert_chain_from_pem_file / https://docs.rs/quiche/latest/quiche/struct.Config.html#method.load_priv_key_from_pem_file (for servers)

It would be great to document these prominently somewhere, as well as clarify the existing documentation which says
> When set to a non-zero value quiche will only allow at most v bytes of incoming stream data to be buffered for each locally-initiated bidirectional stream.
> The default value is 0.

That misleadingly implies that quiche will allow an unlimited number of bytes when it's set to 0, which is not the case; instead it allows no bytes.

### PR DESCRIPTION
**Pull request by LPardue**
Closes #1250

### CODE REVIEW

File: quiche/src/lib.rs [hunk 1]
```diff
@@ -949,6 +953,10 @@ impl Config {
     /// application) and will allow more data to be received as the buffer is
     /// consumed by the application.
     ///
+    /// When set to zero, either explicitly or via the default, quiche will not
+    /// give any flow control to the peer, preventing if from sending any stream
```
Comment (jyn514) [inline quiche/src/lib.rs:957]:
```suggestion
    /// give any flow control to the peer, preventing it from sending any stream
```
(and the same for all the following comments)

Comment (LPardue) [reply quiche/src/lib.rs:957]: derp, thanks

**Review by jyn514 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -69,17 +69,45 @@ production)
...
+### Configuring connections
 
 The first step in establishing a QUIC connection using quiche is creating a
-configuration object:
+[`Config`] object:
 
 ```rust
-let config = quiche::Config::new(quiche::PROTOCOL_VERSION)?;
+let mut config = quiche::Config::new(quiche::PROTOCOL_VERSION)?;
+config.set_application_protos(&[b"h3"]);
```
Comment (ghedo) [inline README.md:79]: This overview is supposed to be somewhat generic, so using the h3 ALPN here makes it sound like everyone has to use it regardless of what protocol runs on top of QUIC. Maybe change it to `example-proto` or something?

File: README.md [hunk 3]
```diff
@@ -69,17 +69,45 @@ production)
...
+
+// additional configuration for application or use case
```
Comment (ghedo) [inline README.md:81]:
```suggestion
// Additional configuration specific to application and use case...
```

File: README.md [hunk 4]
```diff
@@ -69,17 +69,45 @@ production)
...
+
```
Comment (ghedo) [inline README.md:82]:
```suggestion
```

**Review by ghedo — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (LPardue) [reply README.md:79]: yep

**Review by ghedo — APPROVED** [review]

**Pull request merged**